### PR TITLE
Make sure we log error while getting child and data

### PR DESCRIPTION
**Pull request by hiteshk25**

### CODE REVIEW

**Comment by dragonsinth** [comment]
@patsonluk 
:wave: hey Scott! I got a QQ, do u know in what case will we run into this `ErrClosing` ? https://github.com/fullstorydev/gosolr/blame/master/solrmonitor/zkwatcherman.go#L217 we are investigating an issue that some collections seem to lose the child watches (while the data watches are still active) . I couldn't quite reproduce it but am going through some code logic and want to understand a bit more :slightly_smiling_face: https://app.shortcut.com/fullstory/story/200336/solrman-seems-not-able-to-read-correct-replica-states-from-zk-when-there-are-move-split-failures (edited) 

@dragonsinth 
That should only happen when the app is trying to exit.
Basically… once you call `zk.(*Conn).Close()`, calls on that conn start returning that error.
And the `Close()` call is probably in a `defer` somewhere when exiting.

@patsonluk 
ah ok makes sense, that's why i cannot reproduce it. I tried to have a go routine just repeatedly get children and then shut the zk down (`services --hard stop zookeeper`) but it didn't throw such `ErrClosing`

@dragonsinth 
You should never get that error based on something like a network error, or the remote server is restarting
:+1:

@patsonluk 
Thank you! think I can eliminate that as a possible culprit from the investigation
:heart:

**Comment by dragonsinth** [comment]
IOW, I'm not sure this change provides any value; it will simply make all shutdowns very spammy.

**Comment by dragonsinth** [comment]
I'm almost certain that error is ONLY triggered when you close the ZK client, not if the server restarts or if there's a network error.

You can test this out without landing this commit on master; you can temporarily change the gosolr version to a specific commit internally, even if that commit is on a branch.

**Comment by hiteshk25** [comment]
> I'm almost certain that error is ONLY triggered when you close the ZK client, not if the server restarts or if there's a network error.
> 
> You can test this out without landing this commit on master; you can temporarily change the gosolr version to a specific commit internally, even if that commit is on a branch.

Agree I don't see any evidence for it ! 

Then by looking code, it seems zk server drops watches if those are registered during shutdown.
